File: Week_02/G20190343020253/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num = dict()
        for i in range(0, len(nums)):
            a = nums[i]
            b = target - a
            if b in num:
                return [num[b], i]
            else:
                num[a] = i

File: Week_01/G20190343020121/two-sum.java
import java.util.HashMap;
import java.util.Map;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int compent = target - nums[i];
            if (map.containsKey(compent)) {
                return new int[]{ map.get(compent), i};
            }
            map.put(nums[i] , i) ;
        }
        throw  new IllegalArgumentException("没有解法");


    }
}